sound/pci/hda/Kconfig

```diff
@@ -226,6 +226,68 @@ config SND_HDA_POWER_SAVE_DEFAULT
 	  The default time-out value in seconds for HD-audio automatic
 	  power-save mode.  0 means to disable the power-save mode.
 
+if SND_HDA_INTEL
+
+# The options below should not be enabled by distributions or
+# users. They are selected by Intel/Skylake or SOF drivers when they
+# register for a PCI ID which is also handled by the HDAudio legacy
+# driver. When this option is selected and the DSP is detected based on
+# the PCI class/subclass/prog-if, the probe of the HDAudio legacy
+# aborts. This mechanism removes the need for distributions to use
+# blacklists. It can be bypassed with module parameters should the
+# Intel/Skylake or SOF drivers fail to handle a specific platform.
+
+config SND_HDA_INTEL_DSP_DETECTION_SKL
+	bool
+	help
```

Comment:
wouldn't it be better to hide these Kconfig entries by removing their help entries, if they are anyway not for humans to toggle?

Reply:
There are different opinions on this. I used comments in the past but was asked to add a help message.

Reply:
I also don't like the fact that commit messages are out of scope in github. Thanks for the typo report.